core/src/main/java/bisq/core/dao/node/parser/TxOutputParser.java

```diff
@@ -171,17 +244,118 @@ private void handleUnlockBondTx(TempTxOutput txOutput) {
+                        return tempTxOutput.getIndex() >= 1;
+                    }
+                    break;
+                case COMPENSATION_REQUEST:
+                    break;
+                case REIMBURSEMENT_REQUEST:
+                    break;
+                case BLIND_VOTE:
+                    // TODO
+                    break;
+                case VOTE_REVEAL:
+                    break;
+                case LOCKUP:
+                    break;
+                case ASSET_LISTING_FEE:
+                case PROOF_OF_BURN:
+                    // Asset listing fee and proof of burn tx are structurally the same.
+
+                    // We need to require one BSQ change output as we could otherwise not be able to distinguish between 2
+                    // structurally same transactions where only the BSQ fee is different. In case of asset listing fee and proof of
+                    // burn it  is a user input, so it is not know to the parser, instead we derive the burned fee from the parser.
+                    // In case of proposal fee  we could derive it from the params.
+
+                    // Case 1: 10 BSQ fee to burn
+                    // In: 17 BSQ
+                    // Out: BSQ change 7 BSQ -> valid BSQ
+                    // Out: OpReturn
+                    // Miner fee: 1000 sat  (10 BSQ burned)
+
+
+                    // Case 2: 17 BSQ fee to burn
+                    // In: 17 BSQ
+                    // Out: burned BSQ change 7 BSQ -> BTC (7 BSQ burned)
+                    // Out: OpReturn
+                    // Miner fee: 1000 sat  (10 BSQ burned)
+                    return tempTxOutput.getIndex() >= 1;
+            }
+        }
+        return false;
+    }
+
+    private boolean isIssuanceCandidateTxOutput(TempTxOutput tempTxOutput) {
+        // If we have BSQ left as fee and we are at the second output we interpret it as a compensation request output.
+        return availableInputValue > 0 &&
+                tempTxOutput.getIndex() == 1 &&
+                optionalOpReturnType.isPresent() &&
+                (optionalOpReturnType.get() == OpReturnType.COMPENSATION_REQUEST ||
+                        optionalOpReturnType.get() == OpReturnType.REIMBURSEMENT_REQUEST);
+    }
+
+    private void handleIssuanceCandidateOutput(TempTxOutput tempTxOutput) {
+        // We do not permit more BSQ outputs after the issuance candidate.
+        prohibitMoreBsqOutputs = true;
+
+        // We store the candidate but we don't apply the TxOutputType yet as we need to verify the fee  after all
+        // outputs are parsed and check the phase. The TxParser will do that....
+        optionalIssuanceCandidate = Optional.of(tempTxOutput);
+    }
+
+    private void handleBlindVoteFeeOutput(TempTxOutput tempTxOutput) {
+        tempTxOutput.setTxOutputType(TxOutputType.BTC_OUTPUT);
+        prohibitMoreBsqOutputs = true;
+    }
+
+    private boolean isBlindVoteFeeOutput(TempTxOutput tempTxOutput) {
```

Comment:
I don't understand the naming of this method

Reply:
Is isBlindVoteBurnedFeeOutput more clear? It it the output of the burned BSQ